test/_certificates/gen-end_user-bad-certificates.sh

```diff
@@ -0,0 +1,56 @@
+#! /bin/bash
+set -e
+
+usage()
+{
+  echo "Usage: ./gen-end_user-certificates.sh [user_suffix]"
+}
+
+if [ -z "$1" ]; then
+   usage
+   exit 1
+fi
+
+mkdir -p user/invalid/expired
+mkdir -p user/invalid/unproper-key-usage
+
+user_name=user-$1
+
+echo "Generating self-signed certificate for user '$user_name'"
+
+openssl genpkey -outform DER -pkeyopt rsa_keygen_bits:4096 -algorithm RSA -out user/invalid/$user_name.priv_key -quiet
+
+# generates a certification request
+openssl req -new \
+  -subj "/C=IN/ST=Mumbai/O=Polygon Technology/OU=Nightfall Team/CN=$user_name/emailAddress=$user_name@polygon.technology" \
```

Comment:
Just remember to remove references to Polygon (also on the email, etc)

Reply:
Very good spot @signorecello!